libs/ng-strapi-auth/src/lib/core/services/user/user.service.ts

```diff
@@ -1,9 +1,44 @@
 import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { map } from 'rxjs/operators';
+import { AuthService } from '../auth/auth.service';
+import { Observable } from 'rxjs';
+import { UserModel } from '../../models';
 
 @Injectable({
-  providedIn: 'root'
+    providedIn: 'root'
 })
 export class UserService {
 
-  constructor() { }
+    private authApiBase: string = 'http://localhost:1337';
+
+    constructor(
+        private httpClient: HttpClient,
+        public authService: AuthService,
+    ) {
+
```

Comment:
here we should subscribe to authStateChanged from auth service and call setUser / removeUser methods when needed